From 639820fdeaa70f0248c2f8665565ecb321b964ff Mon Sep 17 00:00:00 2001
From: Steph de Los Estados Unidos <admin@hanfubk.com>
Date: Wed, 29 Mar 2017 21:59:49 -0400
Subject: [PATCH 1/3] Create 99-aprilfool

April fool - a dirty implementation, but should run only once before reboot and run only on 04-01.

The text needed to show is not done so I just put Welcome to the summoner's rift there.
---
 bashrc.d/99-aprilfool | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 bashrc.d/99-aprilfool

diff --git a/bashrc.d/99-aprilfool b/bashrc.d/99-aprilfool
new file mode 100644
index 0000000..ce9af95
--- /dev/null
+++ b/bashrc.d/99-aprilfool
@@ -0,0 +1,13 @@
+# 99-April Fool
+# By Staph. aureus
+# Install a run-once Easter Egg into /tmp in 2017-04-01
+
+if [ $(date +%F) == "2017-04-01" ]; then
+	if [ ! -f /tmp/scat ] ; then
+		echo \#! /bin/bash >/tmp/scat
+		echo "while IFS= read -rn1 a; do printf %s \"\${a:-\$'\n'}\"; sleep \$1; done" >> /tmp/scat
+		echo sleep \$2 >> /tmp/scat
+		chmod +x /tmp/scat
+		echo Welcome to the summoner\'s rift. | /tmp/scat 0.1 0
+	fi
+fi

From fabc05d562c70cf299aa69dec0db8a39c8ffbd43 Mon Sep 17 00:00:00 2001
From: Steph de Los Estados Unidos <admin@hanfubk.com>
Date: Wed, 29 Mar 2017 22:03:55 -0400
Subject: [PATCH 2/3] Rename 99-aprilfool to 99-aprilfools

---
 bashrc.d/99-aprilfool  | 13 -------------
 bashrc.d/99-aprilfools | 13 +++++++++++++
 2 files changed, 13 insertions(+), 13 deletions(-)
 delete mode 100644 bashrc.d/99-aprilfool
 create mode 100644 bashrc.d/99-aprilfools

diff --git a/bashrc.d/99-aprilfool b/bashrc.d/99-aprilfool
deleted file mode 100644
index ce9af95..0000000
--- a/bashrc.d/99-aprilfool
+++ /dev/null
@@ -1,13 +0,0 @@
-# 99-April Fool
-# By Staph. aureus
-# Install a run-once Easter Egg into /tmp in 2017-04-01
-
-if [ $(date +%F) == "2017-04-01" ]; then
-	if [ ! -f /tmp/scat ] ; then
-		echo \#! /bin/bash >/tmp/scat
-		echo "while IFS= read -rn1 a; do printf %s \"\${a:-\$'\n'}\"; sleep \$1; done" >> /tmp/scat
-		echo sleep \$2 >> /tmp/scat
-		chmod +x /tmp/scat
-		echo Welcome to the summoner\'s rift. | /tmp/scat 0.1 0
-	fi
-fi
diff --git a/bashrc.d/99-aprilfools b/bashrc.d/99-aprilfools
new file mode 100644
index 0000000..ce9af95
--- /dev/null
+++ b/bashrc.d/99-aprilfools
@@ -0,0 +1,13 @@
+# 99-April Fool
+# By Staph. aureus
+# Install a run-once Easter Egg into /tmp in 2017-04-01
+
+if [ $(date +%F) == "2017-04-01" ]; then
+	if [ ! -f /tmp/scat ] ; then
+		echo \#! /bin/bash >/tmp/scat
+		echo "while IFS= read -rn1 a; do printf %s \"\${a:-\$'\n'}\"; sleep \$1; done" >> /tmp/scat
+		echo sleep \$2 >> /tmp/scat
+		chmod +x /tmp/scat
+		echo Welcome to the summoner\'s rift. | /tmp/scat 0.1 0
+	fi
+fi

From 211da16807556f25c00d609857d15c9479591404 Mon Sep 17 00:00:00 2001
From: Steph de Los Estados Unidos <admin@hanfubk.com>
Date: Fri, 31 Mar 2017 01:10:50 -0400
Subject: [PATCH 3/3] Update text

Test this out and see if it might take too long...
---
 bashrc.d/99-aprilfools | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/bashrc.d/99-aprilfools b/bashrc.d/99-aprilfools
index ce9af95..d3df44e 100644
--- a/bashrc.d/99-aprilfools
+++ b/bashrc.d/99-aprilfools
@@ -8,6 +8,20 @@ if [ $(date +%F) == "2017-04-01" ]; then
 		echo "while IFS= read -rn1 a; do printf %s \"\${a:-\$'\n'}\"; sleep \$1; done" >> /tmp/scat
 		echo sleep \$2 >> /tmp/scat
 		chmod +x /tmp/scat
-		echo Welcome to the summoner\'s rift. | /tmp/scat 0.1 0
+		echo " calc:" | /tmp/scat 0.1 0.1
+		echo "   sub ebx, eax" | /tmp/scat 0.1 0.4
+		echo "    jz retv" | /tmp/scat 0.1 0.1
+		echo "    jg calc" | /tmp/scat 0.1 0.1
+		echo "  xchg ebx, eax" | /tmp/scat 0.1 0.4
+		echo "   jmp calc" | /tmp/scat 0.1 0.1
+		echo " retv:" | /tmp/scat 0.1 0.1
+		echo "  ret" | /tmp/scat 0.1 2
+		echo
+		echo "Welcome back, $USER" | /tmp/scat 0.1 0.1
+		echo
+		echo "Major updates are going on with AOSC."
+		echo "Please check the news website (https://aosc.io)"
+		echo "and carefully read all the changes happening now..."
+		sleep 1
 	fi
 fi
